src/main/java/com/bugsnag/BugsnagAppender.java

```diff
@@ -0,0 +1,681 @@
+        }
+    }
+
+    /**
+     * @param apiKey The API key of the appender to get (only required if using multiple API keys)
+     * @return A running instance of the appender (if one has been created)
+     */
+    public static BugsnagAppender getInstance(String apiKey) {
+        if (instances.containsKey(apiKey)) {
+            return instances.get(apiKey);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public void start() {
+        if (apiKey != null && !apiKey.isEmpty()) {
+            this.bugsnag = createBugsnag();
+
+            instances.put(apiKey, this);
+        }
+        super.start();
+    }
+
+    @Override
+    public void stop() {
+        super.stop();
+        if (bugsnag != null) {
+            bugsnag.close();
+            instances.remove(apiKey);
+        }
+    }
+
+    @Override
+    protected void append(final ILoggingEvent event) {
+        if (bugsnag != null) {
+            Throwable throwable = extractThrowable(event);
+
+            // Only send a message if there is an exception
+            if (throwable != null && !detectLogFromBugsnag(throwable)) {
+                bugsnag.notify(
+                        throwable,
+                        calculateSeverity(event),
+                        new Callback() {
+                            @Override
+                            public void beforeNotify(Report report) {
+
+                                // Add some data from the logging event
+                                report.addToTab("Log event data",
+                                        "Message", event.getMessage());
+                                report.addToTab("Log event data",
+                                        "Timestamp", event.getTimeStamp());
+
+                                // Add details from the logging context to the event
+                                populateContextData(report, event);
+                            }
+                        });
+            }
+        }
+    }
+
+    /**
+     * Calculates the severity based on the logging event
+     * @param event the event
+     * @return The Bugsnag severity
+     */
+    private Severity calculateSeverity(ILoggingEvent event) {
+        if (event.getLevel().equals(Level.ERROR)) {
+            return Severity.ERROR;
+        } else if (event.getLevel().equals(Level.WARN)) {
+            return Severity.WARNING;
+        }
+        return Severity.INFO;
+    }
+
+    /**
+     * Checks to see if a stack trace came from the Bugsnag library
+     * (prevent possible infinite reporting loops)
+     *
+     * @param throwable the exception to check
+     * @return true if the stacktrace contains a frame from the Bugsnag library
+     */
+    private boolean detectLogFromBugsnag(Throwable throwable) {
+
+        // Check all places that LOGGER is called with an exception in the Bugsnag library
```

Comment:
I'm not sure how better to protect against this, we can't exclude all of com.bugsnag because we wouldn't get any of the logs from our apps. We should avoid logging with an error message in this code.